File: workers/shared/models/opencopilot_db/chatbot_model.py
from sqlalchemy import Column, String, Boolean, DateTime, Text
from datetime import datetime
import uuid
from .get_declarative_base import Base
from .database_setup import engine
class Chatbot(Base):
    __tablename__ = 'chatbots'

    id = Column(String(36), primary_key=True, default=str(uuid.uuid4()))
    name = Column(String(255), default="My first chatbot")
    website = Column(String(255), default="https://openchat.so")
    status = Column(String(255))  # Assuming ChatbotStatusType is a string-based enum
    prompt_message = Column(Text, nullable=True, default="AI_ASSISTANT_INITIAL_PROMPT")  # Assuming it's a string value
    token = Column(String(255))

    enhanced_privacy = Column(Boolean, default=False)
    smart_sync = Column(Boolean, default=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=True)
    updated_at = Column(DateTime, onupdate=datetime.utcnow, nullable=True)
    deleted_at = Column(DateTime, nullable=True)

Base.metadata.create_all(engine)

File: entity/src/models/chatbot_model.rs
use serde::{Deserialize, Serialize};
use chrono::NaiveDateTime;
#[derive(Deserialize, Serialize)]
pub struct Chatbot {
    pub id: String,
    pub name: String,
    pub token: String,
    pub website: Option<String>,
    pub status: Option<String>,
    pub prompt_message: Option<String>,
    pub enhanced_privacy: Option<i8>,
    pub smart_sync: Option<i8>,
    pub created_at: Option<NaiveDateTime>,
    pub updated_at: Option<NaiveDateTime>,
    pub deleted_at: Option<NaiveDateTime>,
    pub swagger_url: Option<String>,
    pub is_premade_demo_template: Option<i8>,
}
